packages/vue-renderer/src/renderers/base.js

```diff
@@ -0,0 +1,25 @@
+export default class BaseRenderer {
+  constructor(serverContext) {
+    this.serverContext = serverContext
+    this.options = serverContext.options
+
+    this.vueRenderer = this.createRenderer()
+  }
+
+  createRenderer() {
+    throw new Error('`createRenderer()` needs to be implemented')
```

Comment:
This has no coverage as we are internally always implementing these functions. Is it necessary to keep? @clarkdo

Reply:
At least keep the function signature for consistent common definition？

Reply:
JS is not a typed language to expect defining interface. Maybe can be useful for typescript or ts typings. For coverage, we can either ignore the line or write extra internal tests :)

Reply:
Not for type, just for reminding contributors which are gonna implement new renderer.
Yes, could be moved to ts types.